### PR TITLE
Public images

### PR DESCRIPTION
**Pull request by ViacheslavKlimov**
## Pull Request description

Put your PR description here instead of this sentence.   

## General checklist

- [ ] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [ ] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [ ] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [ ] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [ ] Description contains human-readable scope of changes.
- [ ] Description contains brief notes about what needs to be added to the documentation.
- [ ] No merge conflicts, commented blocks of code, code formatting issues.
- [ ] Changes are backward compatible or upgrade script is provided.
- [ ] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [ ] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [ ] If new dependency was added: the dependency tree is checked for conflicts.
- [ ] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [ ] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [ ] If new yml property was added: make sure a description is added (above or near the property).

### CODE REVIEW

File: application/src/main/data/upgrade/3.6.1/schema_update.sql [hunk 1]
```diff
@@ -16,6 +16,28 @@
 
 -- RESOURCES UPDATE START
 
+ALTER TABLE resource ADD COLUMN IF NOT EXISTS descriptor varchar;
+ALTER TABLE resource ADD COLUMN IF NOT EXISTS preview bytea;
+ALTER TABLE resource ADD COLUMN IF NOT EXISTS external_id uuid;
+ALTER TABLE resource ADD COLUMN IF NOT EXISTS is_public boolean default true;
+ALTER TABLE resource ADD COLUMN IF NOT EXISTS public_key varchar(35) unique;
```
Comment (ashvayka) [inline application/src/main/data/upgrade/3.6.1/schema_update.sql:23]: public_resource_key

File: application/src/main/data/upgrade/3.6.1/schema_update.sql [hunk 2]
```diff
@@ -16,6 +16,28 @@
...
+
+CREATE INDEX IF NOT EXISTS idx_resource_etag ON resource(tenant_id, etag);
+CREATE INDEX IF NOT EXISTS idx_resource_image_public_key ON resource(public_key) WHERE resource_type = 'IMAGE';
```
Comment (ashvayka) [inline application/src/main/data/upgrade/3.6.1/schema_update.sql:26]: check

Comment (ashvayka) [reply application/src/main/data/upgrade/3.6.1/schema_update.sql:26]: type + public key

File: application/src/main/java/org/thingsboard/server/controller/ImageController.java [hunk 3]
```diff
@@ -92,7 +94,8 @@ public class ImageController extends BaseController {
     @PreAuthorize("hasAnyAuthority('SYS_ADMIN', 'TENANT_ADMIN')")
     @PostMapping("/api/image")
     public TbResourceInfo uploadImage(@RequestPart MultipartFile file,
-                                      @RequestPart(required = false) String title) throws Exception {
+                                      @RequestPart(required = false) String title,
+                                      @RequestPart(required = false) Boolean isPublic) throws Exception {
```
Comment (ashvayka) [inline application/src/main/java/org/thingsboard/server/controller/ImageController.java:98]: default "true"

File: application/src/main/java/org/thingsboard/server/controller/ImageController.java [hunk 4]
```diff
@@ -138,6 +142,7 @@ public TbResourceInfo updateImageInfo(@PathVariable String type,
                                           @RequestBody TbResourceInfo newImageInfo) throws ThingsboardException {
         TbResourceInfo imageInfo = checkImageInfo(type, key, Operation.WRITE);
         imageInfo.setTitle(newImageInfo.getTitle());
+        imageInfo.setPublic(newImageInfo.isPublic());
```
Comment (ashvayka) [inline application/src/main/java/org/thingsboard/server/controller/ImageController.java:145]: separate call

File: application/src/main/java/org/thingsboard/server/controller/ImageController.java (hunk 3 above)
Comment (ViacheslavKlimov) [reply application/src/main/java/org/thingsboard/server/controller/ImageController.java:98]: no `defaultValue` prop in `@RequestPart`

**Pull request merged**